node/src/db/models/tokens.rs

```diff
@@ -110,4 +134,59 @@ mod test {
 
         Ok(())
     }
+
+    #[actix_rt::test]
+    async fn store_append_only_state() -> anyhow::Result<()> {
+        load_env();
+        let (client, _lock) = test_db_client().await;
+        let mut initial_data: HashMap<&str, Value> = HashMap::new();
+        initial_data.insert("value", json!(true));
+        initial_data.insert("value2", json!(4));
+        let token = TokenBuilder {
+            initial_data_json: json!(initial_data.clone()),
+            ..TokenBuilder::default()
+        }
+        .build(&client)
+        .await?;
+        assert_eq!(json!(initial_data), token.initial_data_json);
+        assert_eq!(json!(initial_data), token.additional_data_json);
+
+        let mut state_data_json: HashMap<&str, Value> = HashMap::new();
+        state_data_json.insert("value", Value::Null);
+        state_data_json.insert("value2", json!(8));
+        state_data_json.insert("value3", json!(2));
+        Token::store_append_only_state(
+            NewTokenAppendOnly {
+                token_id: token.id,
+                state_data_json: json!(state_data_json),
+            },
+            &client,
+        )
+        .await?;
+        let mut expected_data = initial_data.clone();
+        expected_data.insert("value", Value::Null);
+        expected_data.insert("value2", json!(8));
+        expected_data.insert("value3", json!(2));
+        let token = Token::load(token.id, &client).await?;
+        assert_eq!(json!(expected_data), token.additional_data_json);
```

Comment:
cool, just a bit less verbose way:
```suggestion
        let state_data_json = json!({"value": Value::Null, "value2": 8, "value3": 2});
        Token::store_append_only_state(
            NewTokenAppendOnly {
                token_id: token.id,
                state_data_json: state_data_json.clone(),
            },
            &client,
        )
        .await?;
        let token = Token::load(token.id, &client).await?;
        assert_eq!(state_data_json, token.additional_data_json);
```